estuary/ui
# Preserve side panel docs display state

## Issues

The issues directly below are completely resolved by this PR:
https://github.com/estuary/ui/issues/737

## Changes

The following features are included in this PR:

-   737
    * Remove the `show` state property and `setShow` action from the `SidePanelDocsStore`.
    * Store a `boolean` flag in local storage that indicates whether the side panel docs should be displayed.
    * Create a context provider to access and set the aforementioned side panel docs display flag in local storage.

## Tests

Manually tested

Approaches to testing are as follows:

-   737
    * Confirm that the side panel docs open by default when a create/edit workflow is entered (i.e., when they have not been explicitly closed).

    * Close the side panel docs in a create/edit workflow for one entity type, reopen the side panel docs, and validate that the correct documentation is displayed.

    * Close the side panel docs in a create/edit workflow for one entity type, switch connectors at least twice, and validate that the side panel docs remain closed.

    * Close the side panel docs in a create/edit workflow for one entity type, switch connectors at least once, reopen the side panel docs, and validate that the correct documentation is displayed.

    * Close the side panel docs in a create/edit workflow for one entity type, complete the workflow, enter the create/edit workflow for another entity type, and validate that the side panel docs remain closed.

    * Close the side panel docs in a create/edit workflow for one entity type, reopen the side panel docs, complete the workflow, enter the create/edit workflow for another entity type, and validate that the side panel docs remain open and the correct documentation is displayed.

## Screenshots

N/A
